d2common/d2fileformats/d2cof/cof.go

```diff
@@ -193,7 +192,15 @@ func (c *COF) Marshal() []byte {
 
 		sw.PushBytes(byte(c.CofLayers[i].DrawEffect))
 
-		sw.PushBytes(c.CofLayers[i].weaponClassByte...)
+		s := c.CofLayers[i].WeaponClass.String()
+
+		for j := 0; j < 4; j++ {
+			if j < len(s) {
+				sw.PushBytes(s[j])
+			} else {
+				sw.PushBytes(0)
+			}
+		}
```

Comment:
what is this magic 4?

Reply:
the single-letter variables deserve descriptive names

Reply:
right, fixed in 1010353

Reply:
?
```golang
	const (
		maxCodeLength = 3 // we assume item codes to look like 'hax' or 'kit'
		terminator    = 0
	)

	weaponCode := c.CofLayers[i].WeaponClass.String()

	for idx, letter := range weaponCode {
		if idx > maxCodeLength {
			break
		}

		sw.PushBytes(byte(letter))
	}

	sw.PushBytes(terminator)
	```